tests/bluetooth/tester/src/audio/btp_bap_broadcast.c

```diff
@@ -52,12 +54,52 @@ static inline struct bt_bap_stream *stream_broadcast_to_bap(struct btp_bap_broad
 	return &stream->audio_stream.cap_stream.bap_stream;
 }
 
-struct btp_bap_broadcast_local_source *btp_bap_broadcast_local_source_get(uint8_t source_id)
+struct btp_bap_broadcast_local_source *btp_bap_broadcast_local_source_allocate(
+	uint32_t broadcast_id)
 {
-	/* Only one local broadcast source supported for now */
-	(void) source_id;
+	for (uint8_t i = 0; i < ARRAY_SIZE(local_sources); i++) {
+		if (local_sources[i].broadcast_id == broadcast_id) {
+			LOG_ERR("Local source already allocated for broadcast id %d", broadcast_id);
 
-	return &local_source;
+			return NULL;
+		}
+	}
+
+	for (uint8_t i = 0; i < ARRAY_SIZE(local_sources); i++) {
+		if (!local_sources[i].allocated) {
+			local_sources[i].allocated = true;
+			local_sources[i].source_id = i;
+			local_sources[i].broadcast_id = broadcast_id;
+
+			return &local_sources[i];
+		}
+	}
+
+	return NULL;
+}
+
+int btp_bap_broadcast_local_source_free(uint8_t source_id)
+{
+	if (source_id < ARRAY_SIZE(local_sources) && local_sources[source_id].allocated) {
+		memset(&local_sources[source_id], 0, sizeof(local_sources));
+
+		return 0;
+	}
+
+	return -EINVAL;
+}
```

Comment:
Could also have done something like 

```suggestion
int btp_bap_broadcast_local_source_free(struct btp_bap_broadcast_local_source *source)
{
        if (source == NULL) {
            return -EINVAL;
        }
        
        memset(sources, 0, sizeof(*source);
}
```

But either way works